src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java

```diff
@@ -5,110 +5,127 @@
+        Optional responseJson = Optional.empty();
+        try {
+            response = restTemplate.exchange(url, HttpMethod.GET, entity, String.class);
+        } catch (RestClientException e) {
+            log.warn(word + " isn't an English word!");
+            return responseJson;
+        }
 
-        JsonObject responseJson = new JsonParser().parse(response.getBody()).getAsJsonObject();
+        responseJson = Optional.of(new JsonParser().parse(response.getBody()).getAsJsonObject());
         return responseJson;
     }
+
     @Override
-    public boolean checkPartOfSpeech(String word, String expectedPartOfSpeech, JsonObject wordDetails) {
-        JsonArray definitions = wordDetails.getAsJsonArray("results");
-        Set<String> partsOfSpeech = new HashSet<>();
+    public boolean checkPartOfSpeech(String word, String expectedPartOfSpeech) {
+        if(getWordDetails(word).isEmpty()){
+            return false;
+        } else {
+            JsonObject wordDetails = getWordDetails(word).get();
+            JsonArray definitions = wordDetails.getAsJsonArray("results");
+            Set<String> partsOfSpeech = new HashSet<>();
 
-        for (int i = 0; i < definitions.size(); i++) {
-            JsonObject definition = definitions.get(i).getAsJsonObject();
-            partsOfSpeech.add(definition.get("partOfSpeech").getAsString());
+            for (int i = 0; i < definitions.size(); i++) {
+                JsonObject definition = definitions.get(i).getAsJsonObject();
+                try {
+                    partsOfSpeech.add(definition.get("partOfSpeech").getAsString());
+                } catch (Exception e) {
+                    // Words API may occasionally have a null parts of speech. This is an error on their part.
+                    log.warn("Words API parts of speech null. Set for manual review.");
+                    return true;
+                }
+            }
+            return partsOfSpeech.contains(expectedPartOfSpeech);
         }
-
-        return partsOfSpeech.contains(expectedPartOfSpeech);
     }
+
     @Override
     public boolean validatePhraseComponent(String word, String expectedPartOfSpeech) {
-        if ((expectedPartOfSpeech == "noun" || expectedPartOfSpeech == "verb") && word.length() == 0) {
-            Exception e = new IllegalStateException();
-            LOGGER.log(Level.SEVERE, "Critical error: expected component " + expectedPartOfSpeech + " is empty!", e);
-            return false;
-        }
-        Boolean validPartOfSpeech = checkPartOfSpeech(word, expectedPartOfSpeech, getWordDetails(word));
-        try {
-            if (validPartOfSpeech) {
-                return true;
-            } else {
-                LOGGER.warning("The word passed in isn't a " + expectedPartOfSpeech + "!");
-                return false;
-            }
-        } catch (Exception e) {
-            LOGGER.warning("The " + expectedPartOfSpeech + " passed in isn't an English word!");
+
```

Comment:
You don't need this method... In the places where you call this method, you can call checkPartOfSpeech() directly.

Reply:
I'm not sure what method you are referring to.

Reply:
The method validatePhraseComponent(). It just calls checkPartOfSpeech() and returns the return value. You could instead, just call checkPartOfSpeech() wherever you are currently calling validatePhraseComponent().

Reply:
I had to dig in to wrap my brain around this again. I believe the reasoning behind having the validatePhraseComponent() function is to log which word is the problem (or words, if multiple problems). If we just called checkPartOfSpeech() for each && piece of validatePhrase > boolean validPhrase, we get an end result of true or false for the whole phrase. If we want to log any individual words as the problem, we either need to keep validatePhraseComponent() as it is or refactor validatePhrase() to do all the word checks individually instead of in a boolean with multiple &&s. How do you prefer to handle it?

Reply:
Here is a diff describing what I had in mind. (Google 'how to read a diff' for help reading it, if need be)

```jjames@hppav06:~/src/dmpj-folks/mrsbluerose/tribe-app-backend$ gd src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java
diff --git a/src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java b/src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java
index a04c217..467b819 100644
--- a/src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java
+++ b/src/main/java/com/savvato/tribeapp/services/ToBeReviewedCheckerServiceImpl.java
@@ -86,31 +86,25 @@ public class ToBeReviewedCheckerServiceImpl implements ToBeReviewedCheckerServic
                     return true;
                 }
             }
-            return partsOfSpeech.contains(expectedPartOfSpeech);
-        }
-    }
-
-    @Override
-    public boolean validatePhraseComponent(String word, String expectedPartOfSpeech) {
 
-        Boolean validPartOfSpeech = checkPartOfSpeech(word, expectedPartOfSpeech);
+            boolean contains = partsOfSpeech.contains(expectedPartOfSpeech);
 
-        if (validPartOfSpeech) {
-            return true;
-        } else {
-            log.warn("The word passed in isn't a " + expectedPartOfSpeech + "!");
-            return false;
+            if (contains) {
+                return true;
+            } else {
+                log.warn("The word passed in isn't a " + expectedPartOfSpeech + "!");
+                return false;
+            }
         }
-
     }
 
     @Override
     public void validatePhrase(ToBeReviewed tbr) {
 
-        boolean validPhrase = validatePhraseComponent(tbr.getNoun(), "noun")
-                && validatePhraseComponent(tbr.getVerb(), "verb")
-                && (tbr.getAdverb().equals(Constants.NULL_VALUE_WORD) || validatePhraseComponent(tbr.getAdverb(), "adverb"))
-                && (tbr.getPreposition().equals(Constants.NULL_VALUE_WORD) || validatePhraseComponent(tbr.getPreposition(), "preposition"));
+        boolean validPhrase = checkPartOfSpeech(tbr.getNoun(), "noun")
+                && checkPartOfSpeech(tbr.getVerb(), "verb")
+                && (tbr.getAdverb().equals(Constants.NULL_VALUE_WORD) || checkPartOfSpeech(tbr.getAdverb(), "adverb"))
+                && (tbr.getPreposition().equals(Constants.NULL_VALUE_WORD) || checkPartOfSpeech(tbr.getPreposition(), "preposition"));
 
         if (validPhrase) {
             tbr.setHasBeenGroomed(true);

```

Basically, removing the validatePhraseComponent() method, and adding the logic it had which logged the error message into checkPartOfSpeech().

Reply:
Refactored and removed validate Phrase component.